Doc/c-api/typeobj.rst

```diff
@@ -2144,29 +2280,138 @@ and :c:data:`PyType_Type` effectively act as defaults.)
 
 .. c:member:: destructor PyTypeObject.tp_finalize
 
-   An optional pointer to an instance finalization function.  Its signature is::
+   An optional pointer to an instance finalization function.  This is the C
+   implementation of the :meth:`~object.__del__` special method.  Its signature
+   is::
 
       void tp_finalize(PyObject *self);
 
-   If :c:member:`~PyTypeObject.tp_finalize` is set, the interpreter calls it once when
-   finalizing an instance.  It is called either from the garbage
-   collector (if the instance is part of an isolated reference cycle) or
-   just before the object is deallocated.  Either way, it is guaranteed
-   to be called before attempting to break reference cycles, ensuring
-   that it finds the object in a sane state.
+   The primary purpose of finalization is to perform any non-trivial cleanup
+   that must be performed before the object is destroyed, while the object and
+   any other objects it directly or indirectly references are still in a
+   consistent state.  See :pep:`442`.  The finalizer is allowed to execute
```

Comment:
No, don't look at the original change proposal! It's in the See Also section, that's enough.
```suggestion
   consistent state.  The finalizer is allowed to execute
```